tests/shared/utils/test_common.py

```diff
@@ -140,3 +142,47 @@ async def my_function():
             await asyncio.sleep(0)
 
         await my_function()
+
+
+@pytest.mark.parametrize(
+    "module_path, lookup_path, outcome",
+    [
+        ("rasa.shared.core.domain.Domain", None, "Domain"),
+        # lookup in locals()
+        ("Domain", None, "Domain"),
+        # not in locals but in lookup_path
+        ("Event", "rasa.shared.core.events", "Event"),
+        # import errors below
+        ("rasa.shared.core.domain.FunkyDomain", None, None),
+        ("FunkyDomain", None, None),
+        ("FunkyDomain", "rasa.shared.core.domain", None),
+    ],
+)
+def test_class_from_module_path(
+    module_path: Text, lookup_path: Optional[Text], outcome: Optional[Text]
+):
+    if outcome is None:
```

Comment:
should we just create two different tests instead of having this if?

Reply:
+1